src/super_gradients/training/metrics/pose_estimation_metrics.py

```diff
@@ -133,98 +134,128 @@ def reset(self) -> None:
+                predicted_scores=predictions[i].scores,
+                gt_joints=gt_joints[i],
+                gt_bboxes=gt_bboxes[i] if gt_bboxes is not None else None,
+                gt_areas=gt_areas[i] if gt_areas is not None else None,
+                gt_iscrowd=gt_iscrowd[i] if gt_iscrowd is not None else None,
             )
 
     def update_single_image(
         self,
         predicted_poses: Union[Tensor, np.ndarray],
         predicted_scores: Union[Tensor, np.ndarray],
-        groundtruths: Union[Tensor, np.ndarray],
-        gt_bboxes: Union[Tensor, np.ndarray],
-        gt_areas: Union[Tensor, np.ndarray],
-        gt_iscrowd: Union[Tensor, np.ndarray, List[bool]],
+        gt_joints,
+        gt_bboxes,
+        gt_areas,
+        gt_iscrowd,
     ):
-        if len(predicted_poses) == 0 and len(groundtruths) == 0:
+        if len(predicted_poses) == 0 and len(gt_joints) == 0:
             return
         if len(predicted_poses) != len(predicted_scores):
             raise ValueError("Length of predicted poses and scores should be equal. Got {} and {}".format(len(predicted_poses), len(predicted_scores)))
-        if len(groundtruths) != len(gt_areas) != len(gt_bboxes) != len(gt_iscrowd):
-            raise ValueError(
-                "Length of groundtruths, areas, bboxes and iscrowd should be equal. Got {} and {} and {} and {}".format(
-                    len(groundtruths), len(gt_areas), len(gt_bboxes), len(gt_iscrowd)
-                )
-            )
 
-        predicted_poses = convert_to_tensor(predicted_poses, dtype=torch.float32, device=self.device)
-        predicted_scores = convert_to_tensor(predicted_scores, dtype=torch.float32, device=self.device)
+        predicted_poses = convert_to_tensor(predicted_poses, dtype=torch.float32, device="cpu")
+        predicted_scores = convert_to_tensor(predicted_scores, dtype=torch.float32, device="cpu")
+
+        if gt_bboxes is None:
+            gt_bboxes = compute_visible_bbox_xywh(torch.tensor(gt_joints[:, :, 0:2]), torch.tensor(gt_joints[:, :, 2]))
+
+        if gt_areas is None:
+            gt_areas = gt_bboxes[:, 2] * gt_bboxes[:, 3]
+
+        if gt_iscrowd is None:
+            gt_iscrowd = [False] * len(gt_joints)
 
-        gt_keypoints = convert_to_tensor(groundtruths, dtype=torch.float32, device=self.device)
-        gt_areas = convert_to_tensor(gt_areas, dtype=torch.float32, device=self.device)
-        gt_bboxes = convert_to_tensor(gt_bboxes, dtype=torch.float32, device=self.device)
-        gt_iscrowd = convert_to_tensor(gt_iscrowd, dtype=torch.bool, device=self.device)
+        gt_keypoints = convert_to_tensor(gt_joints, dtype=torch.float32, device="cpu")
+        gt_areas = convert_to_tensor(gt_areas, dtype=torch.float32, device="cpu")
+        gt_bboxes = convert_to_tensor(gt_bboxes, dtype=torch.float32, device="cpu")
+        gt_iscrowd = convert_to_tensor(gt_iscrowd, dtype=torch.bool, device="cpu")
```

Comment:
Computing on `cpu` is on purpose right?

Reply:
Yes, it is done intentionally. 
Apparently there is literally no gain in computing such operations on gpu and in some cases it is even slower (because of cpu->gpu transfer)